stdlib/experimental/prometheus/scrape.go

```diff
@@ -0,0 +1,334 @@
+
+	gk, err := groupKey.Build()
+	if err != nil {
+		return nil, err
+	}
+	builder := execute.NewColListTableBuilder(gk, p.administration.Allocator())
+
+	builder.AddCol(flux.ColMeta{
+		Label: "_time",
+		Type:  flux.TString,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "_value",
+		Type:  flux.TFloat,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "_measurement", // data source
+		Type:  flux.TString,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "_field", // promethus metric name
+		Type:  flux.TString,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "url",
+		Type:  flux.TString,
+	})
+
+	builder.AppendValue(0, values.New(p.met.Timestamp.String()))
+	builder.AppendValue(1, values.New(val))
+	builder.AppendValue(2, values.New("prometheus"))
+	builder.AppendValue(3, values.New(p.met.Name))
+	builder.AppendValue(4, values.New(p.url))
+
+	return builder.Table()
+}
+
+func (p *PrometheusIterator) Close() error {
+	return nil
+}
+
+// Get quantiles from summary metric
+func makeQuantiles(m *dto.Metric) map[string]interface{} {
+	fields := make(map[string]interface{})
+	for _, q := range m.GetSummary().Quantile {
+		if !math.IsNaN(q.GetValue()) {
+			fields[fmt.Sprint(q.GetQuantile())] = float64(q.GetValue())
+		}
+	}
+	return fields
+}
+
+// Get buckets from histogram metric
+func makeBuckets(m *dto.Metric) map[string]interface{} {
+	fields := make(map[string]interface{})
+	for _, b := range m.GetHistogram().Bucket {
+		fields[fmt.Sprint(b.GetUpperBound())] = float64(b.GetCumulativeCount())
+	}
+	return fields
+}
+
+// Get labels from metric
+func makeLabels(m *dto.Metric) map[string]string {
+	result := map[string]string{}
+	for _, lp := range m.Label {
+		result[lp.GetName()] = lp.GetValue()
+	}
+	return result
+}
+
+// Get name and value from metric
+func getNameAndValue(m *dto.Metric) map[string]interface{} {
+	fields := make(map[string]interface{})
+	if m.Gauge != nil {
+		if !math.IsNaN(m.GetGauge().GetValue()) {
+			fields["gauge"] = float64(m.GetGauge().GetValue())
+		}
+	} else if m.Counter != nil {
+		if !math.IsNaN(m.GetCounter().GetValue()) {
+			fields["counter"] = float64(m.GetCounter().GetValue())
+		}
+	} else if m.Histogram != nil {
+		if !math.IsNaN(float64(m.GetHistogram().GetSampleCount())) {
+			if !math.IsNaN(float64(m.GetHistogram().GetSampleSum())) {
+				fields = makeBuckets(m)
+				fields["count"] = float64(m.GetHistogram().GetSampleCount())
+				fields["sum"] = float64(m.GetHistogram().GetSampleSum())
+			}
+
+		}
+	} else if m.Summary != nil {
+		if !math.IsNaN(float64(m.GetSummary().GetSampleCount())) {
+			if !math.IsNaN(float64(m.GetSummary().GetSampleSum())) {
+				fields = makeQuantiles(m)
+				fields["count"] = float64(m.GetSummary().GetSampleCount())
+				fields["sum"] = float64(m.GetSummary().GetSampleSum())
+			}
+
+		}
+	} else if m.Untyped != nil {
+		if !math.IsNaN(m.GetUntyped().GetValue()) {
+			fields["value"] = float64(m.GetUntyped().GetValue())
+		}
+	}
+	return fields
+}
```

Comment:
It would be nice to see some unit tests here, I think you could use the `httptest` package to mock a server with a `metrics` endpoint and verify that the source produces the expected `executetest.Table` objects.

Reply:
Yes, test writing is my next step. That's a good suggestion. I was thinking about how to test the metrics endpoint consistently.